crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +339,712 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let transaction = NovalnetSyncTransaction {
+            tid: item.request.connector_transaction_id.clone(),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item.request.get_browser_info()?.get_language()?,
+        };
+        Ok(Self {
+            transaction,
+            custom,
         })
     }
 }
 
-impl TryFrom<RefundsResponseRouterData<RSync, RefundResponse>> for RefundsRouterData<RSync> {
+impl TryFrom<RefundsResponseRouterData<RSync, NovalnetRefundSyncResponse>>
+    for RefundsRouterData<RSync>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: RefundsResponseRouterData<RSync, RefundResponse>,
+        item: RefundsResponseRouterData<RSync, NovalnetRefundSyncResponse>,
     ) -> Result<Self, Self::Error> {
+        match item.response.result.status {
+            NovalnetAPIStatus::SUCCESS => {
+                let refund_id = match item.response.transaction.clone() {
+                    Some(transaction) => Some(transaction.tid),
+                    None => None,
+                }
+                .ok_or_else(missing_field_err("transaction id"))?;
+
+                let transaction_status = match item.response.transaction.clone() {
+                    Some(transaction) => Some(transaction.status),
+                    None => None,
+                }
+                .ok_or_else(missing_field_err("transaction status"))?;
+
+                Ok(Self {
+                    response: Ok(RefundsResponseData {
+                        connector_refund_id: refund_id.to_string(),
+                        refund_status: enums::RefundStatus::from(transaction_status),
+                    }),
+                    ..item.data
+                })
+            }
+            NovalnetAPIStatus::FAILURE => {
+                let response = Err(get_error_response(item.response.result, item.http_code));
+                Ok(Self {
+                    response,
+                    status: enums::AttemptStatus::Failure,
+                    ..item.data
+                })
+            }
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetCancelTransaction {
+    tid: String,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetCancelRequest {
+    pub transaction: NovalnetCancelTransaction,
+    pub custom: NovalnetCustom,
+}
+
+impl TryFrom<&PaymentsCancelRouterData> for NovalnetCancelRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        let transaction = NovalnetCancelTransaction {
+            tid: item.request.connector_transaction_id.clone(),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item.request.get_browser_info()?.get_language()?,
+        };
```

Comment:
Can we pass default language as English if `browser_info` or `language` is not present?